lib/puppet/util/network_device/transport/ssh.rb

```diff
@@ -68,9 +68,12 @@ def connect(&block)
   end
 
   def close
-    @channel.close if @channel
-    @channel = nil
-    @ssh.close if @ssh
+    begin
+      @channel.close if @channel
+      @channel = nil
+      @ssh.close if @ssh
```

Comment:
The network device transport code calls [Net::SSH.start](https://github.com/jorhett/puppet/blob/fixssh/lib/puppet/util/network_device/transport/ssh.rb#L34), and since we don't specify a block, the [`Net::SSH.start` method returns the connection](https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh.rb#L235), where the caller is responsible for closing the connection (really it's a `Net::SSH::Connection::Session` object). I think you are seeing an exception because the channel is being closed twice. The first time is when we call it above on line 72. Then on line 74, we call [`@ssh.close`](https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/connection/session.rb#L116-L121) which closes channels associated with the session, and likely raises an exception because the underlying socket has already been closed.

We could probably drop the call to `@channel.close` and just call `@ssh.close`, and remove the begin/rescue block. 

Another approach would be to use the block form of `Net::SSH.start` and let it close the channel(s) and session. You would need to change `@channel` to not be an instance variable and pass it as an argument to the various methods, e.g. `def process_ssh(channel)`, but that would be a bit more intrusive.